linera-execution/src/evm/revm.rs

```diff
@@ -706,11 +732,19 @@ impl<Runtime: ContractRuntime> CallInterceptorContract<Runtime> {
         context: &mut Ctx<'_, Runtime>,
         inputs: &mut CallInputs,
     ) -> Result<Option<CallOutcome>, ExecutionError> {
-        let contract_address = Address::ZERO.create(0);
-        if inputs.target_address == PRECOMPILE_ADDRESS || inputs.target_address == contract_address
+        // Every call to a contract passes by this function.
+        // Three kinds:
+        // --- Call to the PRECOMPILE smart contract.
+        // --- Call to the EVM smart contract itself
```

Comment:
Do you mean here that when a contract calls itself it goes through `call_or_fail`?

Reply:
Yes.
The function `fn call` of the `Inspector` intercepts all the calls to a function. The first such call is indeed the initial call to the contract, which may be to deploy it or to call a specific function.